### ISSUE TITLE
Gvfs prefetch --hydrate not sparse aware

### ISSUE THREAD

**Issue by judemars** [microsoft/VFSForGit#1632]
Repro:
1) Gvfs sparse --set parentFolder/childA
2) gvfs prefetch --hydrate --folders parentFolder

Expected: 
Prefetch + hydrate everything in parentFolder/childA

Actual: 
Fails because it tries to prefetch + hydrate everything in parentFolder, not just childA subfolder:

Authenticating...Succeeded
Querying remote for config...Succeeded
Configured cache server: Global (xxx)
Checking that GVFS is mounted...Succeeded
Fetching blobs and hydrating files from cache server...Failed. Run 'gvfs log D:\onew7' for more info.

Log files shows a bunch of instances of "Failed to read parentFolder\\childB\\xxxx"

Basically, it seems like gvfs prefetch --hydrate does not limit itself to the cone of folders specified in the sparse set - it tries to prefetch + hydrate everything and fails because the files aren't on disk.

**Comment by arvati** [comment]
The expectation is that the folder paths you specify are relative to the root of the repo, not relative to your current working directory.  I did like this to prefetch one subfolder:
`gvfs prefetch --folders Pictures/2019/ --verbose --hydrate`
